src/smart-components/role/add-role/schema.js

```diff
@@ -161,19 +161,62 @@ export const schemaBuilder = (container) => {
             name: 'add-permissions',
             title: intl.formatMessage(messages.addPermissions),
             StepTemplate: AddPermissionTemplate,
-            nextStep: ({ values }) =>
-              values &&
-              values['add-permissions-table'] &&
-              values['add-permissions-table'].some(({ uuid }) => uuid.split(':')[0].includes('cost-management'))
-                ? 'cost-resources-definition'
-                : 'review',
+            nextStep: ({ values }) => {
+              if (
+                values &&
+                values['add-permissions-table'] &&
+                values['add-permissions-table'].some(({ uuid }) => uuid.split(':')[0].includes('inventory'))
+              ) {
+                return 'inventory-groups-role';
+              }
+              if (
+                values &&
+                values['add-permissions-table'] &&
+                values['add-permissions-table'].some(({ uuid }) => uuid.split(':')[0].includes('cost-management'))
+              ) {
+                return 'cost-resources-definition';
+              }
+
+              return 'review';
+            },
             fields: [
               {
                 component: 'add-permissions-table',
                 name: 'add-permissions-table',
               },
             ],
           },
+          {
+            title: intl.formatMessage(messages.inventoryGroupsAccessTitle),
+            name: 'inventory-groups-role',
+            nextStep: ({ values }) => {
+              if (
+                values &&
+                values['add-permissions-table'] &&
+                values['add-permissions-table'].some(({ uuid }) => uuid.split(':')[0].includes('cost-management'))
```

Comment:
You can simplify this by using safe accessor
```suggestion
                values?.['add-permissions-table']?.some(({ uuid }) => uuid.split(':')[0].includes('cost-management'))
```